### ISSUE TITLE
Docs - ``too-many-locals`` - message description is not helpful

### ISSUE THREAD

**Issue by orSolocate** [PyCQA/pylint#8196]
### Bug description

The current description for `too-many-locals` message is unclear at:
https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/too-many-locals.html

`Used when a function or method has too many local variables.`

Issue opened continuing [this SO discussion](https://stackoverflow.com/questions/63579502/what-is-pylint-too-many-local-variables-trying-to-tell-me-and-what-do-i-do-wit).

### Configuration

```ini
N/A
```

### Command used

```shell
N/A
```

### Pylint output

```shell
N/A
```

### Expected behavior

Suggested description:

`Raised when a function or method has too many local variables, which can imply that it can be difficult to follow, maintain, read or test it.`

Maybe a good example in `good/bad.py` could also help users understand this message instead of just raising the threshold for this message or ignore it.

### Pylint version

```shell
2.17.0 dev docs
```

### OS / Environment

N/A

### Additional dependencies

N/A

**Comment by orSolocate** [comment]
Do you guys agree that the description is vague? we can discuss what the proper message should be, and I could then upload a correction including and `good/bad.py` example and maybe an explanation of how to configure the `.rc` file to change the threshold for this message

**Comment by Pierre-Sassoulas** [comment]
Agree ! This is part of #5953. Not included in #5953 is talking more about why a message is raised (provide a rational) and less about when it's raised. In this case there's probably a LOT of literature to justify not having too much locale variable that we can quote and add in related.rst.
